packages/react/src/components/Popover/Popover.tsx

```diff
@@ -176,12 +176,43 @@ const arrowPseudoElement = {
   name: 'ArrowPseudoElement',
   fn(data: MiddlewareState) {
     const { elements, rects, placement } = data;
-    const arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
-    const arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+
+    let arrowX = 0;
+    let arrowY = 0;
+
+    if (placement.includes('bottom') || placement.includes('top')) {
+      /* It will always be the same height away when top or bottom */
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+
+      /* default `arrowX` to center of reference, if popover is wider */
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+
+      /* if reference is wider, set arrow in the middle of the floating */
+      if (rects.reference.width > rects.floating.width) {
+        arrowX = rects.floating.width / 2;
+      }
+    } else if (placement.includes('left') || placement.includes('right')) {
+      /* It will always be the same width away when right or left */
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+
+      /* default `arrowY` to center of reference, if popover is taller */
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+
+      /* if reference is taller, set arrow in the middle of the floating */
+      if (rects.reference.height > rects.floating.height) {
+        arrowY = rects.floating.height / 2;
+      }
+    }
```

Comment:
Sorry I am nitpicking, but I find it hard to follow the code and values defined. `arrowX` and `arrowY` are defined with the same values inside each of the ifs, but in opposite order and contradicting comments.

I also tested without the placement checks, and works as intended still.

```suggestion
    let arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
    let arrowY = rects.reference.height / 2 + rects.reference.y - data.y;

    if (rects.reference.width > rects.floating.width) {
      arrowX = rects.floating.width / 2;
    }

    if (rects.reference.height > rects.floating.height) {
      arrowY = rects.floating.height / 2;
    }
```